File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
# 소수의 정의
# 본인과 1 이외의 약수가 아예 없는 숫자
# 루트까지만 보면 된다

from math import sqrt

def isprimenumber(num):
    i = 0
    
    if num == 0:
        return False
    elif num == 1:
        return False
    
    target = int(sqrt(num))
    for i in range(2,target+2):
        if num%i == 0:
            break
        else:
            pass
    if i == target + 1:
        return True
    else:
        return False

# 순열로 볼지 고민중
# 문제는 앞에가 0일 때가 문제

from itertools import permutations

def solution(numbers):
    B = []
    answer = 0
    n = len(numbers)
    N = list(numbers)
    for i in range(1,n+1):
        P = list(permutations(N,i))
        A = []
        for p in P:
            a = ''.join(p)
            A.append(a)
        for a in A:
            if a == str(int(a)):
                if isprimenumber(int(a)) == True and a not in B:
                    B.append(a)
                else:
                    pass
            else:
                pass
        answer = len(B)
    return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function isPrime(num) {
    if (num <= 1) return false;
    if (num === 2) return true;
    if (num % 2 === 0) return false;
    for (let i = 3; i <= Math.sqrt(num); i += 2) {
        if (num % i === 0) return false;
    }
    return true;
}

function getPermutations(arr) {
    const results = new Set();

    const generate = (current, remaining) => {
        if (current.length > 0) {
            results.add(parseInt(current.join('')));
        }
        for (let i = 0; i < remaining.length; i++) {
            if (i > 0 && remaining[i] === remaining[i - 1]) continue;
            generate([...current, remaining[i]], remaining.slice(0, i).concat(remaining.slice(i + 1)));
        }
    };

    generate([], arr);
    return Array.from(results);
}

function solution(numbers) {
    let answer = 0;
    const numberList = numbers.split('');
    
    const uniqueNumbers = getPermutations(numberList);
    
    uniqueNumbers.forEach(num => {
        if (isPrime(num)) answer++;
    });

    return answer;
}

